src/databricks/labs/remorph/helpers/file_utils.py

```diff
@@ -67,16 +67,18 @@ def dir_walk(root: Path):
         yield from dir_walk(each_dir)
 
 
-def get_sql_file(input_path: str | Path) -> list:
+def get_sql_file(input_path: str | Path) -> list[Path]:
     """
     Generator that yields the names of all SQL files in the given directory tree.
     :param input_path: Path
     :return: List of SQL files
     """
+    filenames = []
     for _, _, files in dir_walk(Path(input_path)):
         for filename in files:
             if is_sql_file(filename):
-                yield filename
+                filenames.append(filename)
+    return filenames
```

Comment:
Why we are not using yield? We could have used type Generator[Path, None, None]. If it is required to remove yield, we should update the comments also.

Reply:
yield is returning the `Generator` which is not the expected return type. 

`error:The return type of a generator function should be "Generator"or one of its supertypes `
`for filename in get_sql_file(self.input_path):` 

the above line expects only `Path`, not `Generator Type` so changed it to `list[Path]`. 

Will update the Doc string.